src/net.h

```diff
@@ -334,6 +334,13 @@ class V1TransportSerializer : public TransportSerializer {
     void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) const override;
 };
 
+struct CNodeOptions
```

Comment:
nit: (feel free to ignore) If this is defined inside `CNode`, then callers would use `CNode::Options` instead of `CNodeOptions` and inside the class it would be just `Options` instead of `CNodeOptions`. Also, if/when `CNode` is renamed to something else, it would not be necessary to rename `CNodeOptions` too.

Reply:
Unfortunately there's a gcc/clang bug that means this doesn't work right:

```c++
class X
{
public:
    struct Y
    {
        int foo = 42;
    };

    int SomeFunc(Y y = {});
};
```

gives:

```
$ g++ -std=c++17 -Wall -W -c -o lame.o lame.cpp
lame.cpp:11:25: error: could not convert ‘<brace-enclosed initializer list>()’ from ‘<brace-enclosed initializer list>’ to ‘X::Y’
   11 |     int SomeFunc(Y y = {});
      |                         ^
      |                         |
      |                         <brace-enclosed initializer list>

$ clang -std=c++17 -Wall -W -c -o lame.o lame.cpp
lame.cpp:11:25: error: default member initializer for 'foo' needed within definition of enclosing class 'X' outside of member functions
    int SomeFunc(Y y = {});
                        ^
lame.cpp:7:13: note: default member initializer declared here
        int foo = 42;
            ^
1 error generated.
```

Moving `Y` outside of `X` fixes this. https://stackoverflow.com/a/53423881/ has more details.

Reply:
I see. Omitting ` = {}` from `SomeFunc()` definition seems to resolve this, but then callers need to pass `{}` at the end if they don't wish to pass optional arguments. Or if all arguments are passed via the Options variable, then it will be ok.